.github/workflows/python-app.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
```

Comment:
Is installation via `pip` not ... deprecated?

I seem to remember that it raises a 'use virtual environments' and other warnings and one should instead use system packages (which is not a problem, just use `apt install -y python3-pytest-flake8`)?

Reply:
Hmm, from the action it seems that they still endorse `pip`?
But on lots of CI at work I know I removed all the pip stuff.